Makefile.rules

```diff
@@ -226,9 +236,51 @@ GAP_CXXFLAGS += $(CXXFLAGS)
 ########################################################################
 # Linker flags
 ########################################################################
-GAP_LDFLAGS = $(ABI_CFLAGS)
 
-# Add flags for dependencies
+
+#
+# Determine flags for building kernel extensions for use in GAC
+#
+# For now we basically support GCC and clang, on Linux/macOS/Windows,
+# and as a side effect also some other systems such as FreeBSD.
+# Support for other compilers or operating systems will require some
+# extra tweaks.
+#
+ifneq (,$(findstring cygwin,$(host_os)))
+  GAC_CFLAGS =
+  GAC_LDFLAGS = -shared -Wl,$(GAP_BIN_DIR)/bin/$(GAPARCH)/gap.dll
+  # FIXME: the above won't be correct after "make install"; but then I am not
+  # sure we care about "make install" on Cygwin, so I am not going to work on
+  # this until someone specifically asks for it (and even then I'll require
+  # that person to help test it)
+
+  # also add this, from GAP's LDFLAGS????
+  # -Wl,--stack -Wl,16777216 -Wl,--allow-multiple-definition
+
+# gcc  -shared  src/*.o
+#    -Wl,/home/appveyor/gap/bin/x86_64-pc-cygwin-default64-kv8/gap.dll
+#    -o bin/x86_64-pc-cygwin-default64-kv8/.libs/datastructures.dll
+#    -Wl,--enable-auto-image-base
+#    -Wl,--out-implib,bin/x86_64-pc-cygwin-default64-kv8/.libs/datastructures.dll
```

Comment:
I wonder if we need to / ought to add any of these flags; perhaps @ChrisJefferson has some insights?

In either case, we'll have to test the resulting kernel extensions carefully on cygwin. Just because the trivial kext of `mockpkg` works doesn't mean more complex ones will, too...

Reply:
From GAP's LDFLAGS, the --stack shouldn't be necessary (only the top file needs that). --allow-multiple-definition is often necessary for things compilers get wrong -- I haven't tried without it for a few years so it might be improved, but it also doesn't hurt (it just lets you define the same function multiple times).

Reply:
Just double checked, --stack only does something when building an executable, it doesn't do anything to DLLs (it doesn't hurt either).

Reply:
Any idea about the other options, `--enable-auto-image-base` and `--out-implib`? I looked a the manpage explanations for them, but really couldn't tell... I guess the fact that the Windows CI test pass suggest not using them is fine, but just in case there is a hidden pittrap you might by chance be aware of (don't waste time researching it deeply, though, we'll just gotta try it